lib/membrane/video_compositor/scene/object.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Membrane.VideoCompositor.Scene.Object do
+  @moduledoc """
+  This module holds common types for different kinds of Objects available.
+  """
+
+  alias Membrane.VideoCompositor.Scene.Object.{InputVideo, Layout, Texture}
+
+  @typedoc """
+  Objects are renderable entities in VC, that can serve as input for other
+  objects or as an output of the video.
+
+  They are either Texture structs, structs defining Layouts
+  following Layout.t() definition or InputVideo structs.
+  """
+  @type t :: Layout.t() | Texture.t() | InputVideo.t()
+
+  @typedoc """
+  Defines how an object can be referenced in Scene.
+
+  Objects can be assigned to names and identified
+  at other objects as inputs based on assigned names
+  """
+  @type name :: tuple() | atom()
```

Comment:
Since it will be kept in `Term` in Rust, which implements `hash` and `equals`, AFAIK we can even use `any()` type here (@jerzywilczek correct me if I'm wrong). 
```suggestion
  @type name :: any()
```

Reply:
I would still leave this type restricted. Especially that it's identical in `Membrane.Core` https://hexdocs.pm/membrane_core/Membrane.Element.html#t:name_t/0